components/Input/Input.tsx

```diff
@@ -234,20 +272,27 @@ const InputWrapper = styled('div', {
       true: {
         opacity: 0.7,
         color: '$inputDisabledText',
-
-        '&:hover': {
-          '&::before': {
-            backgroundColor: 'inherit',
-          },
-          '&::after': {
-            backgroundColor: 'inherit',
+        '@hover': {
+          '&:hover': {
+            '&::before': {
+              backgroundColor: 'inherit',
+            },
+            '&::after': {
+              backgroundColor: 'inherit',
+            },
           },
         },
       },
     },
     state: {
       invalid: {
-        color: '$inputInvalidBorder',
+        '@hover': {
+          '&:hover': {
+            '&::after': {
+              backgroundColor: '$inputInvalidBorder',
```

Comment:
Awesome!

I don't know if you saw the resolved conversation I had with Josep on this point in the [mockup](https://app.zeplin.io/project/61497d608f9b6e598e281a34/screen/61497eadad40de21822f2aac)

I told him I was discarding the implementation, but I really like the result, I'm totally approving :+1: